### PR TITLE
Integrate status-console-client into chatapi service

### PR DESCRIPTION
**Pull request by adambabik**
### Todo

- [x] fix the service so that the account is not needed immediately
- [x] enable PFS in `Messanger` from status-console-client
- [ ] ~~test the service and provide examples of JSON-RPC method calls~~ this is not possible without changing `status` service; it's not possible to properly select an account using JSON-RPC. It's possible only using bindings.

### CODE REVIEW

File: services/chatapi/api.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+	contacts, err := api.s.messenger.Contacts()
+	if err != nil {
+		return result, fmt.Errorf("failed to load contacts: %v", err)
+	}
+
+	contactToChat := make(map[client.Contact][]*protocol.Message)
+
+	for _, c := range contacts {
+		messages, err := api.s.messenger.Messages(c, 0)
```
Comment (dshulyak) [inline services/chatapi/api.go:50]: we will load whole db into memory. i think we need specific api to count unread messages per chat and load last msg per chat

File: services/chatapi/service.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+			Service:   NewPrivateAPI(s),
+			Public:    true,
+		},
+	}
+}
+
+// Start is run when a service is started.
+func (s *Service) Start(server *p2p.Server) error {
+	// get the current account's private key
+	chatAccount, err := s.accountManager.SelectedChatAccount()
```
Comment (dshulyak) [inline services/chatapi/service.go]: i think account is selected after node is started, i had this problem with wallet service

File: services/chatapi/api.go (hunk 1 above)
Comment (adambabik) [reply services/chatapi/api.go:50]: Yes, I will change that later.

File: services/chatapi/service.go (hunk 2 above)
Comment (adambabik) [reply services/chatapi/service.go]: How did you solve it?

Comment (dshulyak) [reply services/chatapi/service.go]: i started part of app that requires key (address and decrypted db in my case)  when account is selected. the same way as for pfs

**Comment by adambabik** [comment]
```
$ nc -U /Users/adamb/repos/status-go/test-dir/geth.ipc
{"jsonrpc":"2.0","method":"status_login", "params": [{"address": "REDACTED", "password": "REDACTED"}], "id":1}
{"jsonrpc":"2.0","id":1,"result":null}
{"jsonrpc":"2.0","method":"status_start", "params": [], "id":1}
{"jsonrpc":"2.0","id":1,"result":null}
{"jsonrpc":"2.0","method":"status_joinPublicChat", "params": ["status"], "id":1}
{"jsonrpc":"2.0","id":1,"result":null}
{"jsonrpc":"2.0","method":"status_chats", "params": [], "id":1}
{"jsonrpc":"2.0","id":1,"result":{"unviewed-messages-count":0,"chats":{}}}
```

One problem here is that when a public chat is joined, it should be returned in `status_chats`. Currently, a Whisper subscription is created and messages are loaded but the contact is not added and not returned. This needs to be fixed.

Issue: https://github.com/status-im/status-console-client/issues/99

**Comment by adambabik** [comment]
It's outdated. We have https://github.com/status-im/status-protocol-go now.

**Pull request closed without merging**